### PR TITLE
[CH-154] Pass clickhouse backend config path from gluten to clickhouse backend

### PR DESCRIPTION
**Pull request by taiyang-li**
## What changes were proposed in this pull request?
Add config: `spark.gluten.sql.columnar.backend.ch.config.file` as config file path of clickhouse backend
Like this
```
substrait plan:advanced_extensions {
  enhancement {
    [type.googleapis.com/substrait.Expression] {
      literal {
        map {
          key_values {
            key {
              string: "spark.gluten.sql.columnar.backend.ch.config.file"
            }
            value {
              string: "/data1/liyang/cppproject/spark/spark-3.1.1-bin-hadoop2.7/ch/conf/config.xml"
            }
          }
          key_values {
            key {
              string: "spark.sql.execution.arrow.maxRecordsPerBatch"
            }
            value {
              string: "32768"
            }
          }
        }
      }
    }
  }
}

```

Related PR: https://github.com/Kyligence/ClickHouse/pull/118

### CODE REVIEW

File: jvm/src/main/scala/io/glutenproject/GlutenPlugin.scala [hunk 1]
```diff
@@ -144,6 +144,11 @@ private[glutenproject] object GlutenPlugin {
       nativeConfMap.put(
         GlutenConfig.HIVE_EXEC_ORC_COMPRESS, conf.get(GlutenConfig.SPARK_HIVE_EXEC_ORC_COMPRESS))
     }
+    if (conf.contains(GlutenConfig.GLUTEN_CLICKHOUSE_CONFIF_FILE)) {
+      nativeConfMap.put(
+        GlutenConfig.GLUTEN_CLICKHOUSE_CONFIF_FILE,
```
Comment (jinchengchenghh) [inline jvm/src/main/scala/io/glutenproject/GlutenPlugin.scala:149]: I think the config file priority should be lower than spark conf, the spark conf in runtime scope should have the highest priority. Do you think so?

Comment (taiyang-li) [reply jvm/src/main/scala/io/glutenproject/GlutenPlugin.scala:149]: I agree with you. It's similiar in clickhouse,  session settings > user settings > config settings.

**Comment by lgbo-ustc** [comment]
This PR is related to [Support loading setting from config file and improve logging](https://github.com/Kyligence/ClickHouse/pull/118)

File: jvm/src/main/scala/io/glutenproject/GlutenPlugin.scala (hunk 1 above)
Comment (lgbo-ustc) [reply jvm/src/main/scala/io/glutenproject/GlutenPlugin.scala]: Now we have update the implementation to setup configurations for clickhouse backend.
please to see [ kyligence/clickhouse#118](https://github.com/Kyligence/ClickHouse/pull/118)

File: shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala [hunk 2]
```diff
@@ -261,6 +261,7 @@ object GlutenConfig {
   // For ClickHouse Backends.
   val GLUTEN_CLICKHOUSE_SEP_SCAN_RDD = "spark.gluten.sql.columnar.separate.scan.rdd.for.ch"
   val GLUTEN_CLICKHOUSE_SEP_SCAN_RDD_DEFAULT = true
+  val GLUTEN_CLICKHOUSE_CONFIF_PREFIX = "spark.gluten.sql.columnar.backend.ch"
```
Comment (jinchengchenghh) [inline shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala:264]: GLUTEN_CLICKHOUSE_CONFIF_PREFIX 
CONFIF -> CONFIG

**Comment by jinchengchenghh** [comment]
Please modify the PR header

**Review by jinchengchenghh — APPROVED** [review]

**Comment by taiyang-li** [comment]
> 

done.

**Pull request merged**